src/graphics/vertex-format.js

```diff
@@ -75,28 +79,41 @@ Object.assign(pc, function () {
         this.hasColor = false;
         this.hasTangents = false;
         this._defaultInstancingFormat = null;
+        this.verticesByteSize = 0;
+        this.vertexCount = vertexCount;
+        this.interleaved = !vertexCount;
 
-        // calculate total size
+        // calculate total size of the vertex
         this.size = description.reduce(function (total, desc) {
             return total + Math.ceil(desc.components * _typeSize[desc.type] / 4) * 4;
         }, 0);
 
-        var offset = 0;
+        var offset = 0, elementSize;
         for (i = 0, len = description.length; i < len; i++) {
             var elementDesc = description[i];
+
+            // align up the offset to elementSize (when vertexCount is specified only - case of non-interleaved format)
+            elementSize = elementDesc.components * _typeSize[elementDesc.type];
+            if (vertexCount)
+                offset = pc.math.roundUp(offset, elementSize);
+
             element = {
                 name: elementDesc.semantic,
-                offset: elementDesc.hasOwnProperty('offset') ? elementDesc.offset : offset,
-                stride: elementDesc.hasOwnProperty('stride') ? elementDesc.stride : this.size,
+                offset: (vertexCount ? offset : (elementDesc.hasOwnProperty('offset') ? elementDesc.offset : offset)),
+                stride: (vertexCount ? elementSize : (elementDesc.hasOwnProperty('stride') ? elementDesc.stride : this.size)),
                 stream: -1,
                 scopeId: graphicsDevice.scope.resolve(elementDesc.semantic),
                 dataType: elementDesc.type,
                 numComponents: elementDesc.components,
                 normalize: (elementDesc.normalize === undefined) ? false : elementDesc.normalize,
-                size: elementDesc.components * _typeSize[elementDesc.type]
+                size: elementSize
             };
             this.elements.push(element);
-            offset += Math.ceil(element.size / 4) * 4;
+
+            if (vertexCount)
```

Comment:
Sorry, for being picky, but bracket-less conditions are not within styling guidelines.

Reply:
thanks, I'll keep an eye on these.

Reply:
fixed this case.